experiments/prediction/environments/RandomWalk.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from RlGlue import BaseEnvironment
+
+# Constants
+LEFT = 0
+RIGHT = 1
+
+class RandomWalk(BaseEnvironment):
+    def __init__(self, size=5):
+        self.states = size
+        self.state = size // 2
+
+    def start(self):
+        self.state = self.states // 2
+        return self.state
+
+    def step(self, a):
+        if a == LEFT:
+            self.state = max(self.state - 1, -1)
+
+        elif a == RIGHT:
+            self.state = min(self.state + 1, self.states)
+
+        reward = 0
+        terminal = False
+
+        if self.state == -1:
+            reward = -1
+            terminal = True
+
+        elif self.state == self.states:
+            reward = 1
+            terminal = True
+
+        return (reward, self.state, terminal)
+
+    def getXPRD(self, target, rep):
+        N = self.states
+        # build the state * feature matrix
+        # add an extra state at the end to encode the "terminal" state
+        X = np.array([
+            rep.encode(i) for i in range(N + 1)
+        ])
+
+        # build a transition dynamics matrix
+        # following policy "target"
+        P = np.zeros((N + 1, N + 1))
+        pl, pr = target.probs(0)
+        P[0, 1] = pr
+        P[0, N] = pl
+        P[N - 1, N - 2] = pl
+        P[N - 1, N] = pr
+        for i in range(1, N - 1):
+            P[i, i - 1] = pl
+            P[i, i + 1] = pr
+
+        # build the average reward vector
+        R = np.zeros(N + 1)
+        R[0] = pl * -1
+        R[N-1] = pr
+
+        D = np.diag([0.111111, 0.222222, 0.333333, 0.222222, 0.111111, 0])
+
+        return X, P, R, D
+
+class Inverted:
```

Comment:
Maybe call these Inverted_features, Tabular_features, etc?
Or maybe a comment saying that this function creates the inverted features for the random walk environment.

Reply:
Good idea. I don't like the `_features` partially because (a) classes should be title-case and not snake-cases and (b) it's too much information so it will clutter up the code.

I'm thinking maybe `TabularRep` as an alternative. Also with a comment above them that describes what the rep looks like:
```
# Generates a tabular representation:
# [[ 1, 0, 0, 0, 0],
#  [ 0, 1, 0, 0, 0],
#  [ 0, 0, 1, 0, 0],
#  [0, 0, 0, 0, 1],
#  [0, 0, 0, 0, 1]]
```
and perhaps a citation to Sutton 2009